doc/x-interface.md

```diff
@@ -0,0 +1,105 @@
+  - The error flag included in the response channel indicates processing errors encountered by the accelerator.
+    The actions to be taken by a core to recover from accelerator errors are not yet defined.
+
+#### Response Transaction
+The response channel is handshaked according to the following scheme:
+- The initiator asserts `p_valid`. The assertion of `p_valid` must not depend on `p_ready`. The assertion of ready may depend on `p_valid`.
+- Once `p_valid` has been asserted all data must remain stable.
+- The receiver asserts `p_ready` whenever it is ready to receive the transaction. Asserting `p_ready` by default is allowed. While `p_valid` is low, `p_ready` may be retracted at any time.
+- When both `p_valid` and `p_ready` are high the transaction is successful.
+
+## Operand Origin
+The operands forwarded to the accelerator interconnect are determined in the same way as any regular RISC-V instructions.
+Any source registers from the integer register file of the offloading core as defined in the [RISC-V User-Level ISA specification](https://riscv.org/wp-content/uploads/2017/05/riscv-spec-v2.2.pdf#page=24) are allowed.
+If source registers are used, operand A, B and C contain `rs1`, `rs2` and `rs3` respectively.
+For ternary instructions, the instruction format R4-type instruction format is to be used, defining the `rs3` register address by bits `instr_data[31:27]`.
+
+## Write-back Destination
+The default writeback destination for offloaded instruction is the RISC-V destination register specified by `instr_data[11:7]`.
+
+## Dual-Writeback Instructions
+Custom ISA extensions may mandate dual register writebacks.
+In order to accomodate that need we provision the possibility to reserve multiple destination registers for a single offloaded instruction.
+For even destination registers other than `X0`,  `Xn` and `Xn+1` are reserved for write-back upon offloading a dual write-back instruction, where `Xn` denotes the destination register addresss extracted from `instr_data[11:7]`.
+
+For responses resulting from dual-writeback instructions, the accelerator asserts `p_dualwb`.
+Upon accepting the accelerator response, the offloading core writes back data contained in `p_data0` to register `p_id[4:0]`.
+`p_data1` is written back to `p_id[4:0]` + 1.
+
+
+In order to maximize benefits from dual-writeback instructions, the interconnect response path must accomodate simultaneous transfer of two operation results (`p_data0` and `p_data1`).
+If none of the connected accelerators implement dual write-back instructions, the according signal paths will be removed by synthesis tools.
+
+In order to support accelerators implementing dual write-back instructions, the offloading core must include provisions to reserve two destination registers upon offloading an instruction.
+Also, the core should include provisions for simultaneous writeback, implying dual write-ports to the internal register file.
+
+## Write-after-Write Hazards (NEW)
```

Comment:
In Snitch we do not off-load the instruction when we see a WAW hazard.

Reply:
Yes, I know. 
The "NEW" was a marker for me that I forgot to remove...

Reply:
Thanks a lot @zarubaf!
I'll try to clarify the ordering concept.
On the dual-writeback functionality I fully agree. It will be an optional feature.